leakpro/attacks/mia_attacks/HSJ.py

```diff
@@ -124,52 +132,66 @@ def run_attack(self:Self) -> CombinedMetricResult:
+                                                    self.clip_min,
+                                                    self.clip_max,
                                                     )
 
-        _ , perturbation_distances_out = self.signal(model = shadow_model,
-                                                data_loader = shadow_test_loader,
-                                                norm = self.norm,
-                                                y_target = self.y_target,
-                                                image_target = self.image_target,
-                                                initial_num_evals = self.initial_num_evals,
-                                                max_num_evals = self.max_num_evals,
-                                                stepsize_search = self.stepsize_search,
-                                                num_iterations = self.num_iterations,
-                                                gamma = self.gamma,
-                                                constraint = self.constraint,
-                                                batch_size = self.batch_size,
-                                                verbose = self.verbose,
-                                                clip_min = self.clip_min,
-                                                clip_max = self.clip_max,
-                                                logger = self.logger
+        _ , perturbation_distances_out = self.signal( shadow_model,
+                                                attack_out_dataloader,
+                                                self.logger,
+                                                self.norm,
+                                                self.y_target,
+                                                self.image_target,
+                                                self.initial_num_evals,
+                                                self.max_num_evals,
+                                                self.stepsize_search,
+                                                self.num_iterations,
+                                                self.gamma,
+                                                self.constraint,
+                                                self.batch_size,
+                                                self.verbose,
+                                                self.clip_min,
+                                                self.clip_max,
                                                 )
-        sanity_check = True
-        if sanity_check:
-            self.sanity_check(perturbation_distances_in, perturbation_distances_out)
+
         np.save("perturbation_in.npy", perturbation_distances_in)
         np.save("perturbation_out.npy", perturbation_distances_out)
 
 
+        if self.reproducing_paper_results:
+            self.sanity_check(perturbation_distances_in, perturbation_distances_out)
+
```

Comment:
Why is the normalizations on line 208 needed? They dont do anything?

Reply:
I think the thresholds are off. You create the thresholds based on the non-transformed signals but then you use the transformed signals for thresholding. Why do we need to even transform the signals if it is done this way?